psi4/driver/driver_cbs.py

```diff
@@ -701,13 +936,170 @@ def return_energy_components():
     VARH.update(cfour_psivar_list())
     return VARH
 
+
 VARH = return_energy_components()
 
 
+def _get_default_xtpl(nbasis, xtpl_type):
+    """ A helper function to determine default extrapolation type.
+
+    Parameters
+    ----------
+    nbasis : int
+        Number of basis sets
+    xtpl_type : {'scf', 'corl'}
+        Extrapolation type: 'scf' for the total energy, 'corl' for just the
+        correlation component.
+
+    Returns
+    -------
+    function
+        Extrapolation function to be used.
+    """
+
+    if nbasis == 1:
+        return xtpl_highest_1
+    elif xtpl_type == "scf":
+        if nbasis == 2:
+            return scf_xtpl_helgaker_2
+        elif nbasis == 3:
+            return scf_xtpl_helgaker_3
+        else:
+            raise ValidationError("Wrong number of basis sets supplied to scf_xtpl: %d" % nbasis)
+    elif xtpl_type == "corl":
+        if nbasis == 2:
+            return corl_xtpl_helgaker_2
+        else:
+            raise ValidationError("Wrong number of basis sets supplied to corl_xtpl: %d" % nbasis)
+
+
+def _validate_cbs_inputs(cbs_metadata):
+    """ A helper function which validates the ``cbs_metadata`` format,
+    expands basis sets, and provides sensible defaults for optional arguments.
+
+    Parameters
+    ----------
+    cbs_metadata : array
+        array of dicts containing CBS stage keywords.
+
+    Returns
+    -------
+    array
+        Array of dictionaries, with each item consisting of an extrapolation
+        stage. All validation takes place here.
+    """
+
+    metadata = []
+    for item in cbs_metadata:
+        # 1a) all items must have wfn
+        if "wfn" not in item:
+            raise ValidationError("Stage {d} doesn't have defined level of theory!".format(cbs_metadata.index(item)))
+    # 1b) all items must have basis set
+        if "basis" not in item:
+            raise ValidationError("Stage {d} doesn't have defined basis sets!".format(cbs_metadata.index(item)))
+    # 2a) process required stage parameters and assign defaults
+        stage = {}
+        stage["wfn"] = item["wfn"].lower()
+        stage["basis"] = _expand_bracketed_basis(item["basis"].lower())
```

Comment:
doesn't this need `molecule=molstr` as second arg to fn call? Can probably replace the str with dict nowadays.

Reply:
It doesn't need it (as it builds the basis for H2 when no molecule is provided), but it might be safer to pass it on.

Reply:
Fixed now.